lib/checks/forms/autocomplete-appropriate.js

```diff
@@ -45,8 +45,15 @@ if (axe.commons.text.autocomplete.stateTerms.includes(purposeTerm)) {
 }
 
 const allowedTypes = allowedTypesMap[purposeTerm];
+
+/**
+ * Note:
+ * Firefox returns `node.type` as `text` for `type='month'`
+ * Hence using `node.getAttribute('type')` over `node.type`
+ */
+const nodeType = node.hasAttribute('type') ? node.getAttribute('type') : 'text';
```

Comment:
I agree that we need to use the attribute here rather than the DOM property, but you'll have to normalise it. I don't know exactly what that involves, you can look that up in the spec. It's probably case insensitive. It also needs to default to "text" when there's an invalid value here, not just when type isn't set. Otherwise an element with `type="foobar"` wouldn't be allowed to have autocomplete allowed for text.